modules/postgresql/read_replica.tf

```
@@ -104,6 +104,12 @@ resource "google_sql_database_instance" "replicas" {
       zone = lookup(each.value, "zone", local.zone)
     }
 
+    dynamic "data_cache_config" {
+      for_each = var.edition == "ENTERPRISE_PLUS" && var.data_cache_enabled ? ["cache_enabled"] : []
+      content {
+        data_cache_enabled = var.data_cache_enabled
+      }
+    }
```

How about doing like this ?

```suggestion
    dynamic "data_cache_config" {
      for_each = lookup(each.value, "edition", var.edition) == "ENTERPRISE_PLUS" && lookup(each.value, "data_cache_enabled", var.data_cache_enabled)  ? ["cache_enabled"] : []
      content {
        data_cache_enabled = lookup(each.value, "data_cache_enabled", var.data_cache_enabled)
      }
    }

```

It just requires to add  `data_cache_enabled = optional(bool)` in the variable `read_replicas`.
That way the configuration could be more granular :)

---

That's a good suggestion, thanks!

---

Thanks!
You just need to add the option within variables.tf
```
variable "read_replicas" {
  description = "List of read replicas to create. Encryption key is required for replica in different region. For replica in same region as master set encryption_key_name = null"
  type = list(object({
    name                  = string
    name_override         = optional(string)
     ....
    data_cache_enabled = optional(bool)
  }))
  default = []
}
```